cms/datasets/models.py

```diff
@@ -1,58 +1,150 @@
+        for item in results:
+            item["published"] = published
+
         return results
 
     def fetch_api_response(self, url: str | None = None, params: Mapping | None = None) -> dict:
-        api_response: dict = super().fetch_api_response(url=url, params=params)
+        # Add Authorization header if token is set
+        headers = dict(self.http_headers) if self.http_headers else {}
+        if self.token:
+            # Note: Don't prepend "Bearer " - it's already in the token
+            headers["Authorization"] = self.token
+
+        if url is None:
+            url = self.base_url
+        if params is None:
+            params = {}
+
+        is_detail_request = url.startswith(ONSDataset.Meta.detail_url.split("%s", maxsplit=1)[0])
+
+        try:
+            response = requests.get(url, params=params, headers=headers, timeout=30)
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code == HTTPStatus.TOO_MANY_REQUESTS:
+                logger.warning("Rate limit exceeded when fetching datasets", extra={"url": url})
+                raise
+            # Check for 5xx server errors (500-599)
+            if e.response.status_code >= HTTPStatus.INTERNAL_SERVER_ERROR:
+                logger.error(
+                    "Server error when fetching datasets",
+                    extra={"url": url, "status_code": e.response.status_code},
+                )
+                raise
+            raise
+        except requests.exceptions.RequestException as e:
+            logger.error("Request failed when fetching datasets", extra={"url": url, "error": str(e)})
+            raise
+
+        api_response: dict = response.json()
+        if is_detail_request:
+            api_response = self._process_detail_response(api_response)
+        else:
+            api_response["published"] = params.get("published", "").lower() != "false"
+
         # The dataset API returns the per page count as "count" and the total results as "total_count"
         # Queryish expects "count" to be the total results count, so override it here
         if count := api_response.get("total_count"):
             api_response["count"] = count
+
         return api_response
 
+    def _process_detail_response(self, response: dict) -> dict:
+        # For detail responses, we may need to adjust the structure.
+        # This only applies to the detail URL which currently uses the old format.
+        if response.get("current"):
+            # If it's a versioned dataset response using the old format, convert to new format
+            dataset_data = convert_old_dataset_format(response["current"])
+            # Store the next version info if available (this becomes our unpublished version)
+            next_entry = response.get("next")
+            dataset_data["next"] = convert_old_dataset_format(next_entry) if next_entry else {}
+            return dataset_data
```

Comment:
Expect `current` to not exist, but that `next` may (on the very first publication of an edition).

We need to return either `next` if getting unpublished or `current`/top level if fetching publishing.